R/load_deathrates.R

```diff
@@ -0,0 +1,63 @@
+#' Load deathrates
+#'
+#' Taking the input of a country or iso3c code, load the deathrates in
+#' the format required by [malariasimulation](https://github.com/mrc-ide/malariasimulation/).
+#'
+#' @param iso3c Country code to subset deathrates by.
+#' @param country Country name to subset deathrates by.
+#' @param year Year to subset deathrates by. Can be single value or vector.
+#' If format = 'asmr', year data is given in 5 year chunks. If format = 'nmr',
+#' year is available by single year. There is no difference in the distribution
+#' of values between the two formats.
+#' @param format Format to return deathrates by ('asmr' = by years divisible by
+#' 5, or 'nmr' = single year values). See 'peeps::merge_nmr_asmr' documentation
+#' for details.
+#'
+#' @return Data.frame of deathrates for country.
+#' @export
+load_deathrates <- function(iso3c = NULL, country = NULL, year = NULL,
+                            format = "asmr") {
```

Comment:
Apologies if I didn't make this clear, but I think that (for the malaria use cases I can think of), we will always want to have annual death rates. If not, I'm concened we will see some odd five-yearly stepwise artefact of these rates in the outputs.

So I think we can:

1. Preprocess the rates so that they always give annual outputs. I don't know the best way to do this. You could linearly interpolate between the points, or fit a suitable flexible spline to each set (this will have to be checked quite carefully for each set and at the extremes as it's easy to get odd outputs).
2. Then simplify the remaining code given that we will only be returning a single output format. I think a lot of the code will still be useful, just likely moved to the data-raw/ for the pre-processing step

Reply:
Ah, great! Then this solves the asmr/nmr problem I was worried about. Better to not have users specify a format they wouldn't intuitively understand anyways and do all the pre-processing in the shadows. 

About the splines / linear interpolation - if we need to tune the degrees of freedom and minimize the variance, could we do a stability selection?